hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/IterableStorageManager.java

```diff
@@ -51,29 +60,178 @@ public IterableStorageManager() {
+            log.error(
+                    "Failed link management when removing {}; will be unable to"
+                            + " expire all slots for this contract",
+                    key,
+                    irreparable);
+        }
+        return firstContractKey;
+    }
+
+    /**
+     * Updates the given storage slot with the new values resulting from the contract execution.
+     *
+     * @param store Contract storage store
+     * @param newValue The new value for the slot
+     * @param contractNumber The contract number under consideration
+     * @param key The slot key to update
+     */
+    private void updateAccessedValue(
+            @NonNull final ContractStateStore store,
+            @NonNull Bytes newValue,
+            long contractNumber,
+            @NonNull final Bytes key) {
+        try {
+            Objects.requireNonNull(store);
+            Objects.requireNonNull(newValue);
+            Objects.requireNonNull(key);
+            final var slotKey = new SlotKey(contractNumber, key);
+            final var slotValue =
+                    Objects.requireNonNull(store.getSlotValueForModify(slotKey), () -> "Missing key " + slotKey);
+            final var newSlotValue = slotValue.copyBuilder().value(newValue).build();
+            store.putSlot(new SlotKey(contractNumber, key), newSlotValue);
+        } catch (Exception irreparable) {
+            log.error("Failed link management when updating {}", key, irreparable);
+        }
+    }
+
+    /**
+     * Inserts the given key into the slot storage and into the linked list of storage for the given contract.
+     *
+     * @param store Contract storage store
+     * @param firstContractKey The first key in the linked list of storage for the given contract
+     * @param newValue The new value for the slot
+     * @param contractNumber The contract number under consideration
+     * @param newKey The slot key to insert
+     * @return the new first key in the linked list of storage for the given contract
+     */
+    @NonNull
+    private Bytes insertAccessedValue(
+            @NonNull final ContractStateStore store,
+            @Nullable final Bytes firstContractKey,
+            @NonNull final Bytes newValue,
+            long contractNumber,
+            @NonNull final Bytes newKey) {
+        try {
+            Objects.requireNonNull(store);
+            Objects.requireNonNull(newValue);
+            Objects.requireNonNull(newKey);
+            final var newSlotKey = new SlotKey(contractNumber, newKey);
+            final var newSlotValue = SlotValue.newBuilder()
+                    .value(newValue)
+                    .previousKey(null)
+                    .nextKey(firstContractKey)
+                    .build();
+            store.putSlot(newSlotKey, newSlotValue);
+            return newKey;
+        } catch (Exception irreparable) {
+            log.error("Failed link management when inserting {}", newKey, irreparable);
```

Comment:
I agree that the message seems wrong but I think I might want to keep the scope of the try where it is because if any of the lines that could throw exceptions leading up to the `putSlot` are actually issues (null `key` given or `slotKey` does not map to a valid `slotValue`) are not things which can be mitigated so that the `putSlot` would have the correct value.  Best thing to do is bail and log in that case in my opinion.

Reply:
ok